cmd/juju/application/deployer/charm.go

```diff
@@ -263,6 +263,7 @@ func (d *predeployedLocalCharm) PrepareAndDeploy(ctx *cmd.Context, deployAPI Dep
 type localCharm struct {
 	deployCharm
 	curl *charm.URL
+	base corebase.Base
 	ch   charm.Charm
```

Comment:
Do we have anywhere in the code, that checks if the `curl.Series` and the `base` differ? 

Essentially, I'm asking how costly it would be to remove `charm.URL.Series`. It seems like it might be worth at least doing some exploratory research to see what the damage would be even just in the `api` and `cmd` packages.

Reply:
This was discussed between Ian, Heater and I in Riga. I have a list somewhere of every occurrence of `curl.Series` being read and set. 

The low-down is there are very few places it is used. We only really write it in a few places, to construct charm urls in the first place mostly, and the reads are mostly parsing user input

Removing the `Series` field entirely I don't think is possible. It would be a breaking change in a lot of places, bundles for instance.

I also believe we need charm urls to have uniqueness up to the OS.

My plan is to instead remove all reads of Series from charm URL that are anything more than just parsing user input, then replace Series with OS